File: lms-frontend/src/employee/services/locationService.js
// locationService.js
import api from "../../api/axios";

const BASE_URL = "/api/locations";

const getAllLocations = () => {

    return api.get(
        BASE_URL
    );
};

const getLocationById = (locationId) => {

    return api.get(
        `${BASE_URL}/${locationId}`
    );
};

const createLocation = (locationData) => {

    return api.post(
        BASE_URL,
        locationData
    );
};

const updateLocation = (
    locationId,
    locationData
) => {

    return api.put(
        `${BASE_URL}/${locationId}`,
        locationData
    );
};

const deleteLocation = (locationId) => {

    return api.delete(
        `${BASE_URL}/${locationId}`
    );
};

const locationService = {

    getAllLocations,
    getLocationById,
    createLocation,
    updateLocation,
    deleteLocation
};

export default locationService;

File: backend/src/main/java/com/lms_erp/employee/service/LocationService.java
package com.lms_erp.employee.service;

import com.lms_erp.employee.dto.LocationRequest;
import com.lms_erp.employee.dto.LocationResponse;

import java.util.List;

public interface LocationService {

    LocationResponse createLocation(LocationRequest request);

    LocationResponse updateLocation(Long locationId,
                                    LocationRequest request);

    LocationResponse getLocationById(Long locationId);

    List<LocationResponse> getAllLocations();

    void deleteLocation(Long locationId);
}